### ISSUE TITLE
Using Mosparo Over Tor

### ISSUE THREAD

**Issue by Pleepalee** [mosparo/mosparo#256]
Hello, I try to use Mosparo Over TOR, I've it set up on another domain than the main one, I'm experiencing problems such as:
- "Request to access cookie or storage on "<mosparo project css url>" was blocked because we are blocking all third-party torage access requests and content blocking is enabled."
- "CORS header ‘Access-Control-Allow-Origin’ does not match"

Both clearnet and TOR domains on my website are correctly set up in the Mosparo dashboard, however the Mosparo domain is a clearnet one.
Is there something that prevents Mosparo from working over the Tor network ?
Or am I doing something wrong ?

**Comment by zepich** [comment]
Hi @Pleepalee 

Thank you very much for your issue.

In general, I haven't tested mosparo over TOR. But technically, from my perspective, there is no reason why it shouldn't work.

Can you describe what you see right now? Do you see only these errors but no mosparo box in your form? Or does the mosparo box show some error?

Can you try to add the attribute `crossorigin="anonymous"` to the mosparo CSS `<link>` tag and to the `mosparo-frontend.js` `<script>` tag you have on your website?

Thank you very much for your answer.

Kind regards,

zepich

**Comment by Pleepalee** [comment]
Hello, sorry for not answering sooner, I have tried to add it but on the regular version of my website I got this error now: 
"has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource."

**Comment by zepich** [comment]
Hi @Pleepalee 

Don't worry; everything is good :)

Have you added your website's domain as a host in the project settings in mosparo?

![image](https://github.com/user-attachments/assets/41717412-1c6b-4cc6-b253-fda99d6f4edb)

This setting sets the correct CORS headers for the AJAX/XHR requests mosparo needs.

If that does not solve the issue, please let me know which request/file got blocked.

Kind regards,

zepich

**Comment by zepich** [comment]
@Pleepalee Sorry for my last post. I just learned that you need a CORS header for the JS and CSS files if you add `crossorigin="anonymous"` to them.

So on your clearnet domain, you should not need the `crossorigin="anonymous"` attribute at all, and everything should work correctly (when the host setting is configured correctly, but you wrote that in your first post already).

The `crossorigin="anonymous"` header is only needed if you use mosparo from the TOR domain (as far as I understand).

Can you remove the `crossorigin="anonymous" to verify that mosparo would work on the clearnet website?

**Comment by zepich** [comment]
I'm closing this issue because of inactivity. Please feel free to open another issue if the problem still occurs.
